libs/core/src/main/java/org/elasticsearch/common/util/concurrent/AbstractRefCounted.java

```diff
@@ -51,7 +51,12 @@ public final boolean decRef() {
         int i = refCount.decrementAndGet();
         assert i >= 0;
         if (i == 0) {
-            closeInternal();
+            try {
+                closeInternal();
```

Comment:
Can we add javadoc to `closeInternal` to explain that it must handle all exceptions itself, i.e. can never throw?

Reply:
++